### PR TITLE
ICU-21751 Improves failure reporting of the run-with-stubdata CI targ…

### PR DESCRIPTION
**Pull request by gnrunge**
…et and

also updates the instuctions how to reproduce a failure case.

##### Checklist

- [ ] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21751
- [ ] Required: The PR title must be prefixed with a JIRA Issue number. 
- [ ] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: docs/processes/release/tasks/integration.md [hunk 1]
```diff
@@ -143,11 +143,14 @@ should be investigated and fixed.
...
 ```sh
 cd icu4c/source
-cp stubdata/libicudata.so.60.1 lib/
+./runConfigureICU Linux
+make -j2 check
+rm lib/libicudata.so*
+cp -P stubdata/libicudata.so* lib/
 cd test/intltest
 INTLTEST_OPTS=-w make check
 cd ../cintltst
```
Comment (markusicu) [inline docs/processes/release/tasks/integration.md:156]: *after* this line please add something like the following (finishes what the ticket requests)
```

For debugging (for example using gdb) you cannot `make check`. You need to set the `LD_LIBRARY_PATH` for the lib folder, and then run `./cintltst -w` or `./intltest -w` etc. in the debugger.

Example for one possible folder layout of an out-of-source build: `export LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw`
```

Comment (gnrunge) [reply docs/processes/release/tasks/integration.md:156]: Done.

File: .github/workflows/icu_ci.yml [hunk 2]
```diff
@@ -289,13 +289,35 @@ jobs:
...
-          INTLTEST_OPTS=-w make check 2>&1 | grep 'Elapsed Time: '
+          INTLTEST_OPTS=-w make check 2>&1 | tee stubdata_intltest.log;
+          if ! grep 'Elapsed Time: ' stubdata_intltest.log
+          then
+            echo
+            echo intltest run with stubdata failed
+            echo
+            echo See
+            echo https://unicode-org.github.io/icu/processes/release/tasks/integration.html#run-tests-without-icu-data
+            echo how to reproduce and debug the failure
```
[the remark this answers is not in the document]
Comment (gnrunge) [reply .github/workflows/icu_ci.yml]: Done.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**